flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/JSONGenerator.java

```diff
@@ -184,6 +185,8 @@ private void decorateNode(Integer vertexID, ObjectNode node) {
 
         node.put(CONTENTS, vertex.getOperatorDescription());
 
+        node.put(SLOT_SHARING_GROUP, vertex.getSlotSharingGroup());
```

Comment:
Do we ensure the `slotSharingGroup` has been set properly before? If so, we can add a non-null check here.

Reply:
I'm not 100% sure. It's only directly called [here](https://github.com/apache/flink/blob/b4bb9c8bffe1e37ad6912348d8b3bef89af42286/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java#L998) (though it get called indirectly in several other places

<img width="1318" alt="Screen Shot 2022-05-19 at 11 05 22 AM" src="https://user-images.githubusercontent.com/27927454/169369361-a7d0de51-4c73-45ca-b783-0594d67af4e3.png">

From the tests result, I believe it's properly set in both the Table batch and streaming planner. But I can also add the checks for extra safety?

Reply:
Assuming `env.getStreamGraph().getStreamingPlanAsJSON()` is the majority way to get the JSON plan. Then I believe the `slotSharingGroup` is properly set whenever `env.getStreamGraph()` is called. Here is the call stacks for the relevant part

<img width="2398" alt="Screen Shot 2022-05-19 at 11 22 55 AM" src="https://user-images.githubusercontent.com/27927454/169372921-f9233961-2fe9-4197-bb03-b093b3496816.png">

Reply:
Then, we can add a check as a safety net here.

Reply:
Before adding the check, I want to think a little bit through this. I think there're 3 scenarios w/ or w/o check

1. w/ check: if null, throw exception 
2. w/ check: if null, avoid adding the field `slot_sharing_group` 
3. w/o check: ,if null return `"slot_sharing_group": null` in the JSON 

I think I'm lean on 2 & 3. The reason to not throw an exception is that this API locates around the "edge of of the call stacks" - it's printed out to the console as an execution plan & DAG graph and it's likely not be depended by other APIs directly. 3) has a small benefit to maintain a stable JSON schema - it gives a clear message that the value has not been set properly (`null`), while a missing field may requires someone to look into the code to find out that the field is skipped because of `null`.

Reply:
I'm still lean to option 1. The `slotSharingGroup` are by designed to be non-null for a compiled streamGraph. We should proactively through an exception if we get a partially completed streamGraph here.